arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -52,6 +52,32 @@ static void _region_init(u32_t index, struct arm_mpu_region *region_conf)
 		index, region_conf->base, region_conf->attr);
 }
 
+/* ARM Core MPU Driver API Implementation for ARM MPU */
+
+/**
+ * @brief enable the MPU
+ */
+void arm_core_mpu_enable(void)
+{
+	/* Enable MPU and use the default memory map as a
+	 * background region for privileged software access.
+	 */
+	MPU->CTRL = MPU_CTRL_ENABLE_Msk | MPU_CTRL_PRIVDEFENA_Msk;
+
+	/* Make sure that all the registers are set before proceeding */
+	__DSB();
+	__ISB();
+}
+
+/**
+ * @brief disable the MPU
+ */
+void arm_core_mpu_disable(void)
+{
+	/* Disable MPU */
+	MPU->CTRL = 0;
```

Comment:
__DSB()

Reply:
This one is not needed in the current design, because the _disable() is only called in order to re-program the MPU and re-enable it. So it is called in controlled situations.

[Similarly, we do not need a __DSB _before_ enabling the MPU; we only need it _after_]